### PR TITLE
Extend coral plant blocktags

### PR DESCRIPTION
**Pull request by Axionize**
Also add new ALIVE_CORAL_PLANTS and DEAD_CORAL_PLANTS blocktags

### CODE REVIEW

**Comment by booky10** [comment]
These tags don't exist in vanilla, not even in snapshots, why should we add them?

**Comment by Axionize** [comment]
I added the dead states because of falses in Grim due to dead coral not being modeled as coral with the same hitboxes.

I figured someone somewhere would still want the old behavior of only getting alive coral and dead coral. The important thing is to add the dead coral to the corals blocktag.

I wasn't aware blocktags were just ports off a vanilla feature. I'll delete the new tags and just leave this PR to add the dead corals to the coral tags if you're insistent on it.

**Comment by booky10** [comment]
you can still add them, just move them to the bottom of the class - packetevents already has some "inofficial"/"legacy" block tags

everything above the comments is auto-generated and will be replaced with the next version update

i'm also planning on re-designing the item/block tags, as they no longer represent vanilla's tags properly

**Comment by Axionize** [comment]
I'm not sure exactly where you mean by bottom of the class. At the end of the entries or at the actual bottom below the code? what about adding entries to the blocktag?

**Comment by booky10** [comment]
See how GLASS_BLOCKS or GLASS_PANES are defined

**Comment by Axionize** [comment]
Fixed @booky10

File: api/src/main/java/com/github/retrooper/packetevents/protocol/world/states/defaulttags/BlockTags.java [hunk 1]
```diff
@@ -311,7 +319,7 @@ public class BlockTags {
...
         BlockTags.CORAL_BLOCKS.add(StateTypes.TUBE_CORAL_BLOCK, StateTypes.BRAIN_CORAL_BLOCK, StateTypes.BUBBLE_CORAL_BLOCK, StateTypes.FIRE_CORAL_BLOCK, StateTypes.HORN_CORAL_BLOCK);
         BlockTags.WALL_CORALS.add(StateTypes.TUBE_CORAL_WALL_FAN, StateTypes.BRAIN_CORAL_WALL_FAN, StateTypes.BUBBLE_CORAL_WALL_FAN, StateTypes.FIRE_CORAL_WALL_FAN, StateTypes.HORN_CORAL_WALL_FAN);
-        BlockTags.CORAL_PLANTS.add(StateTypes.TUBE_CORAL, StateTypes.BRAIN_CORAL, StateTypes.BUBBLE_CORAL, StateTypes.FIRE_CORAL, StateTypes.HORN_CORAL);
+        BlockTags.CORAL_PLANTS.add(StateTypes.TUBE_CORAL, StateTypes.BRAIN_CORAL, StateTypes.BUBBLE_CORAL, StateTypes.FIRE_CORAL, StateTypes.HORN_CORAL, StateTypes.DEAD_TUBE_CORAL, StateTypes.DEAD_BRAIN_CORAL, StateTypes.DEAD_BUBBLE_CORAL, StateTypes.DEAD_FIRE_CORAL, StateTypes.DEAD_HORN_CORAL);
```
Comment (booky10) [inline api/src/main/java/com/github/retrooper/packetevents/protocol/world/states/defaulttags/BlockTags.java]: The block tags should represent vanilla tags
If you want a tag which contains both alive and dead coral, please add an unofficial tag

**Review by booky10 — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/github/retrooper/packetevents/protocol/world/states/defaulttags/BlockTags.java (hunk 1 above)
Comment (Axionize) [reply api/src/main/java/com/github/retrooper/packetevents/protocol/world/states/defaulttags/BlockTags.java]: should I call it `ALIVE_DEAD_CORAL_PLANTS` `ALIVE_AND_DEAD_CORAL_PLANTS` or `ALL_CORAL_PLANTS` ?

Comment (booky10) [reply api/src/main/java/com/github/retrooper/packetevents/protocol/world/states/defaulttags/BlockTags.java]: `ALL_CORAL_PLANTS` sounds more future-proof

Comment (Axionize) [reply api/src/main/java/com/github/retrooper/packetevents/protocol/world/states/defaulttags/BlockTags.java]: Done.

**Review by booky10 — APPROVED** [review]

**Pull request merged**